lib/private/DB/QueryBuilder/ExpressionBuilder/MySqlExpressionBuilder.php

```diff
@@ -52,4 +54,19 @@ public function iLike($x, $y, $type = null): string {
 		$y = $this->helper->quoteColumnName($y);
 		return $this->expressionBuilder->comparison($x, ' COLLATE ' . $this->collation . ' LIKE', $y);
 	}
+
+	/**
+	 * Returns a IQueryFunction that casts the column to the given type
+	 *
+	 * @param string|IQueryFunction $column
+	 * @param mixed $type One of IQueryBuilder::PARAM_*
```

Comment:
```suggestion
	 * @param IQueryBuilder::PARAM_NULL|IQueryBuilder::PARAM_BOOL|IQueryBuilder::PARAM_INT|IQueryBuilder::PARAM_STR|IQueryBuilder::PARAM_LOB|IQueryBuilder::PARAM_DATE|IQueryBuilder::PARAM_INT_ARRAY|IQueryBuilder::PARAM_STR_ARRAY $type
```

This is a bit more verbose, but nicer for psalm

Reply:
Decided to go for `@psalm-param IQueryBuilder::PARAM_* $type` as it is a bit more flexible and shorter. Psalm would also allow this in `@param` but seems phpstorm does not handle that.